core/chains/evm/txmgr/txmgr.go

```diff
@@ -54,13 +54,13 @@ type Config interface {
 }
 
 // KeyStore encompasses the subset of keystore used by txmgr
-type KeyStore interface {
-	CheckEnabled(address common.Address, chainID *big.Int) error
-	EnabledKeysForChain(chainID *big.Int) (keys []ethkey.KeyV2, err error)
-	GetNextNonce(address common.Address, chainID *big.Int, qopts ...pg.QOpt) (int64, error)
-	GetStatesForChain(chainID *big.Int) ([]ethkey.State, error)
-	IncrementNextNonce(address common.Address, chainID *big.Int, currentNonce int64, qopts ...pg.QOpt) error
-	SignTx(fromAddress common.Address, tx *gethTypes.Transaction, chainID *big.Int) (*gethTypes.Transaction, error)
+type KeyStore[ADDR any, ID any, TX any, META any] interface {
+	CheckEnabled(address ADDR, chainID ID) error
+	//EnabledKeysForChain(chainID ID) (keys []ethkey.KeyV2, err error) //TODO:remove it and follow GetStatesForChain
+	GetNextMetadata(address ADDR, chainID ID, qopts ...pg.QOpt) (META, error)
+	GetStatesForChain(chainID ID) ([]ethkey.State, error)
```

Comment:
Delete this method from here, and replace with the following:

GetEnabledAddressesForChain(chainId ID) ([Addr], error)

Once you do that, you don't even need to worry about the eth.State struct. No need to export that.